core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -19,11 +19,38 @@
 
 package org.apache.iceberg;
 
+import java.util.Set;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+
 public class TableProperties {
 
   private TableProperties() {
   }
 
+  /**
+   * Reserved table property for table format version.
+   * <p>
+   * Iceberg will default a new table's format version to the latest stable and recommended version.
+   * This reserved property keyword allows users to override the Iceberg format version of the table metadata.
+   * <p>
+   * If this table property exists when creating a table, the table will use the specified format version.
+   * If a table updates this property, it will try to upgrade to the specified format version.
+   * This helps developers to try out new features in an unreleased version or migrate existing tables to a new version.
+   * <p>
+   * Note: incomplete or unstable versions cannot be selected using this property.
+   */
+  public static final String RESERVED_PROPERTY_FORMAT_VERSION = "format-version";
```

Comment:
nit: Do we add the `RESERVED_PROPERTY_` prefix to be explicit? I am wondering whether it is worth making the var name longer as we have `RESERVED_PROPERTIES` below.

Just asking. I am fine either way.

Reply:
Yeah that's a good point, I had the prefix because there was no `RESERVED_PROPERTIES` set. But now it becomes a bit redundant.